src/Pate/Arch.hs

```diff
@@ -127,12 +143,80 @@ data ValidArchData arch =
                 --
                 -- For example, these could be PLT stub symbols for ELF binaries
                 , validArchPatchedExtraSymbols :: Map.Map BS.ByteString (BVS.BV (MC.ArchAddrWidth arch))
+                , validArchStubOverrides :: ArchStubOverrides arch
                 }
 
+-- | A PLT stub is allowed to make arbitrary modifications to the symbolic state
```

Comment:
This is currently used for PLT stubs, but could in principle be used for any function that is to be overridden. That doesn't really require any changes, but might be worth a note in the comment.